lib/mix/lib/mix/tasks/format.ex

```diff
@@ -112,6 +156,71 @@ defmodule Mix.Tasks.Format do
     end
   end
 
+  # This function reads exported configuration from the imported dependencies and deals with
+  # caching the result of reading such configuration in a manifest file.
+  defp fetch_deps_opts(formatter_opts) do
+    deps = formatter_opts[:import_deps]
+
+    cond do
+      is_nil(deps) or deps == [] ->
+        formatter_opts
+
+      is_list(deps) ->
+        dep_parenless_calls =
+          if deps_dot_formatters_stale?() do
+            dep_parenless_calls = eval_deps_opts(deps)
+
+            if dep_parenless_calls != [] do
+              write_deps_manifest(dep_parenless_calls)
+            end
+
+            dep_parenless_calls
+          else
+            read_deps_manifest()
+          end
+
+        Keyword.update(
+          formatter_opts,
+          :locals_without_parens,
+          dep_parenless_calls,
+          &(&1 ++ dep_parenless_calls)
+        )
+
+      true ->
+        Mix.raise("Expected :import_deps to return a list of dependencies, got: #{inspect(deps)}")
+    end
+  end
+
+  defp deps_dot_formatters_stale?() do
+    Mix.Utils.stale?([".formatter.exs" | Mix.Project.config_files()], [deps_manifest()])
+  end
+
+  defp deps_manifest() do
+    Path.join(Mix.Project.build_path(), ".cached_deps_formatter.exs")
```

Comment:
`Mix.Project.manifest_path()`

Reply:
It is also best to remove the .exs extension to avoid issues.

Reply:
`.exs` was a residue of initially dumping the Elixir term, good catch. Pushed a fix that uses `manifest_path/0` instead of `build_path/0` as well.